chips/atmega328p-hal/src/wdt.rs

```diff
@@ -0,0 +1,108 @@
+//! Provides a system reset when a counter reaches a given time-out value.
+//!
+//! # Example
+//! ```
+//! let mut watchdog = Wdt::new(&dp.CPU.mcusr, dp.WDT);
+//! watchdog.disable();
+//! watchdog.start(WatchdogTimeOutPeriod::Ms8000);
+//!
+//! loop {
+//!     watchdog.feed();
+//! }
+//! ```
+
+use avr_device::generic::Reg;
+use avr_hal::hal::watchdog::*;
+
+pub enum WatchdogTimeOutPeriod {
+    Ms16,
+    Ms32,
+    Ms64,
+    Ms125,
+    Ms250,
+    Ms500,
+    Ms1000,
+    Ms2000,
+    Ms4000,
+    Ms8000,
+}
+
+pub struct Wdt<'wdt> {
+    mcu_status_register: &'wdt Reg<crate::atmega328p::cpu::mcusr::MCUSR_SPEC>,
+    peripheral: crate::atmega328p::WDT,
+}
+
+impl<'wdt> Wdt<'wdt> {
+    pub fn new(
+        mcu_status_register: &'wdt Reg<crate::atmega328p::cpu::mcusr::MCUSR_SPEC>,
```

Comment:
This will prevent moving `Peripherals.CPU` anyway so I'd just take a reference to the entire peripheral here (`&'wdt crate::atmega328p::CPU`).

Reply:
Do you think it'd be reasonable to clear the WDRF in the MCUSR in `new` instead of clearing it in enable/disable (and only holding onto MCUSR until `new` returns)? It does seem like moving `Peripherals.CPU` into `Wdt` only to reset the flag shouldn't be necessary. Destroying the `Wdt` instance after enabling the watchdog doesn't seem possible since `feed` must be called within each time-out period.

Reply:
> Do you think it'd be reasonable to clear the WDRF in the MCUSR in new instead of clearing it in enable/disable (and only holding onto MCUSR until new returns)?

I've read through the datasheet a bit more and I think this should be ok.  The interactions aren't entirely obvious but unless I'm mistaken, it works like this:

The `WDRF` has two purposes:

1. It signals a starting system that reset was caused by the watchdog.  This is in line with all the other bit-fields in `MCUSR`.
2. It acts as a safeguard to clearing `WDE`.  The idea seems to be that a system which was reset by the watchdog has to 'acknowledge' the watchdog-reset (by clearing `WDRF`) before it can turn off the watchdog.

With both, `WDRF` only becomes relevant in a failure condition and the following restart.  That means, clearing `WDRF` once at startup should be enough as it will only become set when a reset occurs.

This would get rid of the borrowed/owned `CPU` peripheral so it seems to me that this would be the favorable aproach.

Reply:
Thank you, much appreciated.